File: 🟡 Medium/3043 - Find the Length of the Longest Common Prefix/_3043FindtheLengthoftheLongestCommonPrefix.py
class Solution:
    def longestCommonPrefix(self, arr1: List[int], arr2: List[int]) -> int:
        res = 0

        prefixesOfarr1 = set()

        for el in arr1:
            while el not in prefixesOfarr1 and el > 0:
                prefixesOfarr1.add(el)
                el //= 10

        for el in arr2:
            while el not in prefixesOfarr1 and el > 0:
                el //= 10

            if el > 0:
                res = max(res, (int)(log10(el) + 1))

        return res

File: 🟡 Medium/3043 - Find the Length of the Longest Common Prefix/_3043FindtheLengthoftheLongestCommonPrefix.js
/**
 * @param {number[]} arr1
 * @param {number[]} arr2
 * @return {number}
 */
var longestCommonPrefix = function (arr1, arr2) {
    let res = 0;

    const prefixesOfarr1 = new Set();

    for (let el of arr1) {
        while (!prefixesOfarr1.has(el) && el > 0) {
            prefixesOfarr1.add(el);
            el = Math.floor(el / 10);
        }
    }

    for (let el of arr2) {
        while (!prefixesOfarr1.has(el) && el > 0) {
            el = Math.floor(el / 10);
        }

        if (el > 0) {
            res = Math.max(res, Math.floor(Math.log10(el) + 1));
        }
    }

    return res;
};